source/shared/cpp/ObjectModel/BaseElement.cpp

```diff
@@ -0,0 +1,193 @@
+#include "pch.h"
+
+#include "BaseElement.h"
+#include "ParseUtil.h"
+
+namespace AdaptiveSharedNamespace
+{
+    unsigned int InternalId::s_currentInternalId = 1;
+
+    InternalId InternalId::Current()
+    {
+        InternalId internalId{s_currentInternalId};
+        return internalId;
+    }
+
```

Comment:
I think we talked about this once, but can you remind me of the resolution? Could we increment s_currentInternalId here? Do we need to get it more than once from here?

Reply:
We need a way to get the current internal ID without incrementing because we need to be able to push an internal ID on the stack before `BaseElement` is constructed. `Current()` is used in `BaseElement::BaseElement()`, and `Next()` is used in both element and action parser wrappers.

---
In reply to: [257416836](https://github.com/Microsoft/AdaptiveCards/pull/2351#discussion_r257416836) [](ancestors = 257416836)

Reply:
Fair enough.

---
In reply to: [257418724](https://github.com/Microsoft/AdaptiveCards/pull/2351#discussion_r257418724) [](ancestors = 257418724,257416836)